### PR TITLE
Fix "Uncaught ReferenceError: prestashop is not defined"

### PR DESCRIPTION
**Pull request by true0r**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | State has not a lang table. After add state in contact shop, browser generate error "Uncaught ReferenceError: prestashop is not defined" and javascript does not work. 
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Reproduce| Select a state whose name is not English

### CODE REVIEW

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -1555,7 +1555,7 @@ public function getTemplateVarShop()
                 'address2' => $address->address2,
                 'postcode' => $address->postcode,
                 'city' => $address->city,
-                'state' => (new State($address->id_state))->name[$this->context->language->id],
+                'state' => (new State($address->id_state))->name,
```
Comment (maximebiloe) [inline classes/controller/FrontController.php]: Is this working with several languages enabled?

Comment (true0r) [inline classes/controller/FrontController.php]: Yes of course

**Comment by maximebiloe** [comment]
Hello @true0r,

How can I reproduce the issue? I set a state for my store address and don't have any JavaScript error.

Thank you

**Comment by true0r** [comment]
Hello @maximebiloe

```php
echo 'sting'[0];
// s
echo 'строка'[0];
// � (first byte letter)
```
You need to select the stat whose name is not English.

**Comment by true0r** [comment]
Hello @maximebiloe

First byte of a character break a smarty template. But this is not a case when a character are represented as a one byte in UTF-8.

This issue is critical to a new shop on PrestaShop 1.7.

**Comment by antoin-m** [comment]
@maximebiloe State names are not translated and thus `(new State($address->id_state))->name` is a String and not an array.

@true0r Thank you very much for finding the bug and fixing it :+1:

**Pull request merged**